### ISSUE TITLE
UI- Implement new content structure for Facility health services (VAMC)

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#11764]
## Issue Description
The VAMC health services content structure has changed from blobby content to discrete reusable components in Drupal. We need to update the front end to accommodate this new data structure. 

<details>
<summary> Current UI </summary>

![image.png](https://images.zenhubusercontent.com/5d8a47393ac3440001d1a2ed/b0a2f750-f45e-4941-96bc-88d206de7608) 

</details>

[New UI specs](https://zpl.io/VKMnO0W)

<details>
<summary> New UI </summary>

![image](https://user-images.githubusercontent.com/55411834/92267449-eeafdb80-ee9d-11ea-91db-f25f4142ac26.png)

</details>

**Dave has entered test data for University Drive to validate the content. See tests below in Acceptance Criteria**

## Front end tasks
- [x] Update graphQL queries for new data structure
- Ensure the page is styled according to [design spec.](https://zpl.io/VKMnO0W)
  - [x] Remove vertical blue line. 
  - [x] Location and contact information, Appointments, Care we provide headings are H3. 
  - [x] Spacing above H3 = 24

---

Each health service includes the following sections  

   1. Common conditions
   2. Intro text 
   3. Location and contact information in the following order

> Name
> Address
> Interactive link to map
> Hours on separate lines by day
> Additional text information about hours (if available)
> One or more phone numbers with label above interactive phone number 
> One or more email addresses with label above interactive email address 
> Fax number

   4. Appointments section 

> Intro text
> Phone number (optional)
> `Referral needed?` (optional)
> `Walk-ins accepted?` (optional)
> `Online scheduling` (optional)

   5. Care we provide at VA Pittsburgh health care with text and list 
   6. May contain link to detail page, such as` Learn more about... `

---

- Ensure the back end data from API is displayed in local and review instance
- Ensure unit tests are available. 
- Run axe checks using the Chrome or Firefox browser plugin
- Zoom layouts to 400% at 1280px width
- Test with 1 or 2 screen readers
- Navigate using the keyboard only
- Confirm changes are rendered correctly in mobile and desktop

## Data testing
The following tests are based on test data entered for University Drive

1. View location and contact information for University Drive Pharmacy
   - [x] Confirm there are two unique locations: Pharmacy and Prescription refills
   - [x] Unique phone number for each location 
   - [x] Unique hours of operation for each location
          Pharmacy location 
             - Hours = 7:30 a.m. to 9:00 p.m. Monday thru Friday, 7:30 a.m. to 4:00 p.m. Saturday/Sunday 
             - Optional note specifies hours of operation on holidays
          Prescription refill location 
             - Hours = 8:0 a.m. to 4:30 p.m. Monday thru Friday
---
2. View location and contact information for University Drive Recreation and creative arts 
    - [x] No hours of operation have been specified
---
3. View location and contact information for University Drive Registry exams
   - [x] Displays hours of operation as defined at facility level
   - [x] Displays main facility phone number
   - [ ] Displays main facility address with name and email of Coordinator
---
4. View location and contact information for University Drive Rehabilitation and prosthetics
   - [x] Displays `Referral needed: Yes`
   - [x] Displays `Walk ins accepted: Yes`
---
5. View location and contact information for University Drive Emergency services
   - [ ] Displays text note for phone = 911 (`In an emergency`...)
---
6. View location and contact information for University Drive Cardiology
   - [x] Displays hours of operation defined in hours M-F, On Call sat-Sun and additional text for `On call 24/7`
---
7. View location and contact information for University Drive MyHealtheVet coordinator
    - [x] Does not display information for Referrals, Walk-ins or Online scheduling
---
8. View location and contact information for University Drive Nutrition, food and dietary
   - [x] Displays online scheduling widget
---
9. View location and contact information for University Drive Suicide Prevention
   - [ ] Displays Monday's hours of operation as A.M. and P.M. and all other hours of operation as a.m. and p.m.
   - [x] Ensure `Learn more and connect with a care coordinator` is actionable and navigates to [Suicide Prevention page](https://www.va.gov/pittsburgh-health-care/suicide-prevention)
---
10. View location and contact information for University Drive Travel reimbursement 
      - [x] Ensure `Learn more about whether you qualify for the VA Beneficiary Travel program` link is actionable and navigates to VA travel pay reimbursement page
---
11. View location and contact information for University Drive Women's Veteran care
      - [x] Ensure Ensure `Learn more and connect with a care coordinator` is actionable and navigates to [Women's Veteran care page](https://www.va.gov/pittsburgh-health-care/women-veteran-care/)

    
## General acceptance Criteria
- [ ] Designer comments on this issue to approve page styling 
- [ ] No data is lost in implementation. Facility health service location and contact info area still contains:
     - Intro text
     - Location information (including building, suite, wing, room, etc)
     - Link to campus map
     - Main phone number (minimum)
     - Hours of operation
     - Appointments
     - Referral
     - Wait time/satisfaction scores
     - Care we provide at (facility) with text intro and list of services 
- [ ] Changes are rendered correctly in mobile and desktop
- [ ] Name of facility is present and accurate within text
- [ ] Unit tests pass
- [ ] End-to-end tests show 0 violations.
- [ ] All axe checks pass
- [ ] All color contrast checks pass
- [ ] All zoom testing passes
- [ ] All keyboard checks pass 
- [ ] All screenreader checks pass

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
 - Include axe checks, including hidden content
 
</details>

<details> 

 <summary> Run axe checks using the Chrome or Firefox browser plugin (expand)</summary>
 
- Ensure no heading levels are skipped.

- Ensure all buttons and labeled inputs use semantic HTML elements.

- Ensure all buttons, labeled elements and images are identified using HTML semantic markers or ARIA roles.

- Ensure form fields have clearly defined boundaries or outlines.

- Ensure form fields do not use placeholder text. 

- Ensure form fields have highly visible and specific error states. 

</details>

<details> 

 <summary> Test for color contrast and color blindness issues (expand) </summary>
 - All text has appropriate contrast. 

</details>

<details> 

 <summary> Zoom layouts to 400% at 1280px width (expand)</summary> 
 - Ensure readability and usability are supported when zoomed up to 400% at 1280px browser width
 - Ensure no content gets focused offscreen or is hidden from view.
 
</details>

<details> 

 <summary> Test with 1 or 2 screen readers (expand)</summary> 
 
 - Ensure the page includes a skip navigation link. 
 
 - Ensure all links are properly descriptive. 
 
 - Ensure screen reader users can hear the text equivalent for each image conveying information. 

 - Ensure screen reader users can hear the text equivalent for each image button. 

 - Ensure screen reader users can hear labels and instructions for inputs. 
 
 - Ensure purely decorative images are not announced by the screenreader. 
 
</details>

<details> 

 <summary>Navigate using the keyboard only (expand)</summary> 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be reached with the tab key in a logical order. 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure all interactive elements can be reached with the tab key in a logical order 
 
 - Ensure all interactive elements can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure focus is always visible and appears in logical order.
 
 - Ensure each interactive element has visible focus state which appears in logical order.
 

</details>

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by mpelzsherman** [comment]
Can we get some documentation about the new content structure?

**Comment by mpelzsherman** [comment]
Re: cutover plan: per @ethanteague, we will need to add a field to Drupal to indicate which model to use. This is a blocking dependency.

**Comment by mpelzsherman** [comment]
@ethanteague has offered to document the new content model, and work with us to update the GraphQL queries and templates.

**Comment by brangi** [comment]
@mmiddaugh after the initial conversation on this thread https://dsva.slack.com/archives/C0FQSS30V/p1599841939019300?thread_ts=1599841151.017800&cid=C0FQSS30V we seem to be blocked by CMS new content

**Comment by brangi** [comment]
Would be nice to have either mockdata or  documentation about the new structure

**Comment by mpelzsherman** [comment]
@ethanteague has taken care of most of the work in https://github.com/department-of-veterans-affairs/vets-website/pull/14279

**Comment by mpelzsherman** [comment]
@mmiddaugh - we've discovered that the link to the interactive map is missing from the structured content in Drupal:

![Screenshot_9_21_20__6_00_PM](https://user-images.githubusercontent.com/80267/93826088-7e8ead00-fc34-11ea-9616-8c676ee1b549.png)

**Comment by mmiddaugh** [comment]
@mpelzsherman @brangi you can leave the map out of the front end for now

**Comment by mpelzsherman** [comment]
@mmiddaugh - please advise as to the following:

> View location and contact information for University Drive Registry exams
>  - Displays hours of operation as defined at facility level

do we need to repeat the facility hours, or can we just add text that says “See facility hours”?

>  Displays main facility phone number

There's a specific phone number in Drupal for Registry exams.

>  Displays main facility address with name and email of Coordinator

I don't see a name or email of Coordinator defined in Drupal

**Comment by mpelzsherman** [comment]
Got an answer from @rtwell on repeating the main facility hours - YES

**Comment by mpelzsherman** [comment]
> Displays Monday's hours of operation as A.M. and P.M. and all other hours of operation as a.m. and p.m.

The data in Durpal is all lowercase

**Comment by mpelzsherman** [comment]
> Learn more about whether you qualify for the VA Beneficiary Travel program link is actionable and navigates to VA travel pay reimbursement page

That link goes to HealthBenefits/vtp/Beneficiary_Travel.asp, which redirects on production to health-care/get-reimbursed-for-travel-pay/. Would be nice to eliminate the redirect.

**Comment by brangi** [comment]
@mpelzsherman  - https://design.va.gov/content-style-guide/dates-and-numbers should be lowercase according to the design system

**Comment by mmiddaugh** [comment]
<details>
<summary>Snapshots for for design review </summary> 

![VA_Pittsburgh_Health_Care___Pittsburgh_VA_Medical_Center-University_Dr (3)](https://user-images.githubusercontent.com/55411834/94317647-f8c87580-ff43-11ea-9bad-6a863b595949.png)

---

![VA_Pittsburgh_Health_Care___Pittsburgh_VA_Medical_Center-University_Dr](https://user-images.githubusercontent.com/55411834/94317650-f9f9a280-ff43-11ea-8391-5f101f59180b.png)

---

![VA_Pittsburgh_Health_Care___Pittsburgh_VA_Medical_Center-University_Dr (1)](https://user-images.githubusercontent.com/55411834/94317651-f9f9a280-ff43-11ea-8075-fff9232a6bff.png)

---

![VA_Pittsburgh_Health_Care___Pittsburgh_VA_Medical_Center-University_Dr (2)](https://user-images.githubusercontent.com/55411834/94317652-f9f9a280-ff43-11ea-837b-5d2fd31f61a4.png)

</details>

**Comment by mpelzsherman** [comment]
The builds for this PR keep failing with a 502 error, which is preventing review.

**Comment by mmiddaugh** [comment]
[Design feedback documented in PR](https://github.com/department-of-veterans-affairs/vets-website/pull/14304#issuecomment-700809161)

**Comment by mmiddaugh** [comment]
@mpelzsherman We need to double-check the handling for phone numbers when "use facility phone" is enabled. 
<details>
<summary> For example, the Nutrition, food and dietary accordion on Staging renders the phone number as 000-000-0000 </summary>

![image](https://user-images.githubusercontent.com/55411834/95596048-1f89b000-0a0a-11eb-8b8d-66b744c4b1f6.png)
 
</details>

<details>
<summary>The service does not list a phone in production (except for Appointment)</summary>

![image](https://user-images.githubusercontent.com/55411834/95596169-4811aa00-0a0a-11eb-8512-61f92314b66f.png)

</details>

<details>
<summary> Drupal calls for the facility phone number to be displayed </summary>

![image](https://user-images.githubusercontent.com/55411834/95597154-82c81200-0a0b-11eb-92ef-e0ce6d340088.png)


</details>

**Comment by mmiddaugh** [comment]
@rtwell @mpelzsherman checking in on the alignment for the "Holidays" line under hours of operation
<details>
<summary>  Screenshot </summary>

![image](https://user-images.githubusercontent.com/55411834/95597757-4812a980-0a0c-11eb-85e6-27f974e476e3.png)

</details>

**Comment by rtwell** [comment]
I'm not sure about the holiday hours bit—what does that mean, exactly? which holidays does it apply to? everything else is looking pretty great.

**Comment by davidconlon** [comment]
"Holidays" is just free content. 
In this case we were reflecting this:

![image](https://user-images.githubusercontent.com/56046371/96757144-0b976400-13a3-11eb-8843-fd6911ed0b7b.png)

**Comment by mpelzsherman** [comment]
Can we close this ticket and write up new ones to address any remaining issues?
